File: lib/prompts.js
const SYSTEM_PROMPT =
  'Você é um assistente especializado em psicologia clínica. ' +
  'Gere documentos clínicos profissionais, objetivos e tecnicamente precisos em português brasileiro. ' +
  'Use linguagem clínica formal. Baseie-se apenas nas informações da transcrição fornecida. ' +
  "Para dados não mencionados, indique 'Não referido na sessão'.";

const TEMPLATES = {
  anamnese: `Com base na transcrição, elabore uma ANAMNESE PSICOLÓGICA completa.

Paciente: {patient_name} | Data: {session_date} | Psicólogo(a): {psychologist_name}
{extra}

Seções obrigatórias:
1. IDENTIFICAÇÃO
2. QUEIXA PRINCIPAL E DEMANDA
3. HISTÓRIA DO PROBLEMA ATUAL (início, evolução, fatores)
4. HISTÓRICO PESSOAL (infância, desenvolvimento, marcos)
5. HISTÓRICO FAMILIAR (dinâmica, figuras significativas)
6. SAÚDE FÍSICA E MEDICAMENTOS
7. VIDA SOCIAL, OCUPACIONAL E AFETIVA
8. SONO, APETITE E ROTINA
9. OBSERVAÇÕES CLÍNICAS INICIAIS

TRANSCRIÇÃO:
{transcript}`,

  evolucao: `Com base na transcrição, elabore uma EVOLUÇÃO DE SESSÃO.

Paciente: {patient_name} | Data: {session_date} | Psicólogo(a): {psychologist_name}
{extra}

Inclua: estado inicial do paciente, conteúdos trabalhados, intervenções realizadas,
resposta às intervenções, material emergente (defesas, emoções, insights),
encerramento e estado final, observações para próxima sessão.

TRANSCRIÇÃO:
{transcript}`,

  soap: `Com base na transcrição, elabore nota no formato SOAP.

Paciente: {patient_name} | Data: {session_date} | Psicólogo(a): {psychologist_name}
{extra}

S – SUBJETIVO (relatos, queixas, sentimentos verbalizados)
O – OBJETIVO (comportamentos observáveis, apresentação, afeto, estado mental)
A – AVALIAÇÃO (impressão clínica, hipóteses, análise da evolução)
P – PLANO (intervenções, estratégias, próximos passos, encaminhamentos)

TRANSCRIÇÃO:
{transcript}`,

  relatorio: `Com base na transcrição, elabore um RELATÓRIO PSICOLÓGICO.

Paciente: {patient_name} | Data: {session_date} | Psicólogo(a): {psychologist_name}
{extra}

1. IDENTIFICAÇÃO DO PACIENTE
2. MOTIVO DO ENCAMINHAMENTO / OBJETIVO
3. PROCEDIMENTOS UTILIZADOS
4. DADOS COLETADOS E ANÁLISE CLÍNICA
5. SÍNTESE PSICODIAGNÓSTICA
6. CONCLUSÃO E RECOMENDAÇÕES
7. (Espaço para assinatura e CRP)

TRANSCRIÇÃO / DADOS:
{transcript}`,

  declaracao: `Elabore uma DECLARAÇÃO DE COMPARECIMENTO formal.

Paciente: {patient_name} | Data: {session_date} | Psicólogo(a): {psychologist_name}
{extra}

A declaração deve confirmar comparecimento à consulta psicológica, indicar data e horário se mencionados,
ser formal e adequada para apresentação institucional, e ter espaço para assinatura e CRP.

INFORMAÇÕES DA SESSÃO:
{transcript}`,

  resumo: `Com base na transcrição, elabore um RESUMO DA SESSÃO conciso (máx. 400 palavras).

Paciente: {patient_name} | Data: {session_date}
{extra}

Capture: principais temas, questões centrais, intervenções e técnicas usadas,
insights ou mudanças percebidas, pontos de atenção para próxima sessão.

TRANSCRIÇÃO:
{transcript}`,

  hipoteses: `Com base na transcrição, elabore HIPÓTESES CLÍNICAS E TEMAS CENTRAIS.

Paciente: {patient_name} | Data: {session_date} | Psicólogo(a): {psychologist_name}
{extra}

TEMAS CENTRAIS OBSERVADOS
HIPÓTESES CLÍNICAS (CID-11 ou DSM-5 se aplicável — apresentar como hipóteses de trabalho)
MECANISMOS DE DEFESA IDENTIFICADOS
RECURSOS E POTENCIALIDADES DO PACIENTE
ÁREAS DE INVESTIGAÇÃO FUTURA

TRANSCRIÇÃO:
{transcript}`,

  plano_terapeutico: `Com base na transcrição, elabore um PLANO TERAPÊUTICO.

Paciente: {patient_name} | Data: {session_date} | Psicólogo(a): {psychologist_name}
{extra}

OBJETIVOS TERAPÊUTICOS (gerais e específicos, curto e médio prazo)
ABORDAGEM E ESTRATÉGIAS (linha teórica, técnicas previstas)
TEMAS PRIORITÁRIOS PARA TRABALHO
FREQUÊNCIA E MODALIDADE PROPOSTAS
INDICADORES DE PROGRESSO
PRÓXIMOS PASSOS IMEDIATOS

TRANSCRIÇÃO:
{transcript}`,
};

function buildPrompt(type, { transcript, patient_name, session_date, psychologist_name, additional_context }) {
  const tmpl = TEMPLATES[type];
  if (!tmpl) throw new Error(`Tipo inválido: ${type}`);
  return tmpl
    .replace('{transcript}', transcript)
    .replace('{patient_name}', patient_name || 'Não informado')
    .replace('{session_date}', session_date || 'Não informado')
    .replace('{psychologist_name}', psychologist_name || 'Não informado')
    .replace('{extra}', additional_context ? `Contexto adicional: ${additional_context}` : '');
}

module.exports = { SYSTEM_PROMPT, TEMPLATES, buildPrompt };

File: backend/prompts.py
SYSTEM_PROMPT = (
    "Você é um assistente especializado em psicologia clínica. "
    "Gere documentos clínicos profissionais, objetivos e tecnicamente precisos em português brasileiro. "
    "Use linguagem clínica formal. Baseie-se apenas nas informações da transcrição fornecida. "
    "Para dados não mencionados, indique 'Não referido na sessão'."
)

DOCUMENT_PROMPTS = {
    "anamnese": """Com base na transcrição da sessão abaixo, elabore uma ANAMNESE PSICOLÓGICA completa e estruturada.

**Paciente:** {patient_name}
**Data da sessão:** {session_date}
**Psicólogo(a):** {psychologist_name}

Organize em seções:
1. IDENTIFICAÇÃO
2. QUEIXA PRINCIPAL E DEMANDA
3. HISTÓRIA DO PROBLEMA ATUAL (início, evolução, fatores desencadeantes)
4. HISTÓRICO PESSOAL (infância, desenvolvimento, marcos)
5. HISTÓRICO FAMILIAR (dinâmica, figuras significativas, padrões)
6. SAÚDE FÍSICA E MEDICAMENTOS
7. VIDA SOCIAL, OCUPACIONAL E AFETIVA
8. SONO, APETITE E ROTINA
9. OBSERVAÇÕES CLÍNICAS INICIAIS

{additional_context}

TRANSCRIÇÃO DA SESSÃO:
{transcript}""",

    "evolucao": """Com base na transcrição abaixo, elabore uma EVOLUÇÃO DE SESSÃO (nota de evolução psicológica).

**Paciente:** {patient_name}
**Data:** {session_date}
**Psicólogo(a):** {psychologist_name}

Inclua:
- Estado geral do paciente no início da sessão
- Conteúdos e temas trabalhados
- Intervenções realizadas pelo psicólogo
- Resposta do paciente às intervenções
- Material emergente (defesas, emoções, insights)
- Encerramento da sessão e estado final
- Observações para próxima sessão

{additional_context}

TRANSCRIÇÃO:
{transcript}""",

    "soap": """Com base na transcrição abaixo, elabore uma nota no formato SOAP.

**Paciente:** {patient_name}
**Data:** {session_date}
**Psicólogo(a):** {psychologist_name}

**S – SUBJETIVO**
(Relatos do paciente, queixas verbalizadas, sentimentos expressos, perspectiva do paciente)

**O – OBJETIVO**
(Comportamentos observáveis durante a sessão, apresentação, afeto, estado mental, dados objetivos)

**A – AVALIAÇÃO**
(Impressão clínica, hipóteses diagnósticas, análise da evolução, nível de funcionamento)

**P – PLANO**
(Intervenções realizadas, estratégias terapêuticas, próximos passos, encaminhamentos se houver)

{additional_context}

TRANSCRIÇÃO:
{transcript}""",

    "relatorio": """Com base na transcrição abaixo, elabore um RELATÓRIO PSICOLÓGICO estruturado.

**Paciente:** {patient_name}
**Data da avaliação:** {session_date}
**Psicólogo(a) responsável:** {psychologist_name}

O relatório deve conter:
1. IDENTIFICAÇÃO DO PACIENTE
2. MOTIVO DO ENCAMINHAMENTO / OBJETIVO DA AVALIAÇÃO
3. PROCEDIMENTOS UTILIZADOS
4. DADOS COLETADOS E ANÁLISE CLÍNICA
5. SÍNTESE PSICODIAGNÓSTICA
6. CONCLUSÃO E RECOMENDAÇÕES
7. ASSINATURA E CRP

Use linguagem técnica, objetiva e fundamentada. Evite julgamentos de valor.

{additional_context}

TRANSCRIÇÃO / DADOS DA SESSÃO:
{transcript}""",

    "declaracao": """Elabore uma DECLARAÇÃO DE COMPARECIMENTO formal para:

**Paciente:** {patient_name}
**Data da sessão:** {session_date}
**Psicólogo(a):** {psychologist_name}

A declaração deve:
- Confirmar que o paciente compareceu à consulta psicológica
- Indicar data, horário (se mencionado na transcrição) e duração
- Ser formal e adequada para apresentação profissional ou institucional
- Conter espaço para assinatura e CRP

{additional_context}

INFORMAÇÕES DA SESSÃO:
{transcript}""",

    "resumo": """Com base na transcrição abaixo, elabore um RESUMO DA SESSÃO conciso e claro.

**Paciente:** {patient_name}
**Data:** {session_date}

O resumo deve capturar:
- Principais temas abordados
- Questões centrais trazidas pelo paciente
- Intervenções e técnicas utilizadas
- Insights ou mudanças percebidas
- Pontos de atenção para próxima sessão

Seja direto e claro. Máximo de 400 palavras.

{additional_context}

TRANSCRIÇÃO:
{transcript}""",

    "hipoteses": """Com base na transcrição abaixo, elabore um documento de HIPÓTESES CLÍNICAS E TEMAS CENTRAIS.

**Paciente:** {patient_name}
**Data:** {session_date}
**Psicólogo(a):** {psychologist_name}

Inclua:

**TEMAS CENTRAIS OBSERVADOS**
(Padrões recorrentes, conflitos nucleares, demandas implícitas)

**HIPÓTESES CLÍNICAS**
(Hipóteses diagnósticas possíveis, com base em CID-11 ou DSM-5 se aplicável)

**MECANISMOS DE DEFESA IDENTIFICADOS**
(Se observáveis na transcrição)

**RECURSOS E POTENCIALIDADES DO PACIENTE**
(Fatores protetivos identificados)

**ÁREAS DE INVESTIGAÇÃO FUTURA**
(O que precisa ser aprofundado nas próximas sessões)

Apresente como hipóteses de trabalho, não como diagnóstico definitivo.

{additional_context}

TRANSCRIÇÃO:
{transcript}""",

    "plano_terapeutico": """Com base na transcrição abaixo, elabore um PLANO TERAPÊUTICO INICIAL ou de PRÓXIMOS PASSOS.

**Paciente:** {patient_name}
**Data:** {session_date}
**Psicólogo(a):** {psychologist_name}

Estruture em:

**OBJETIVOS TERAPÊUTICOS**
(Gerais e específicos, a curto e médio prazo)

**ABORDAGEM E ESTRATÉGIAS**
(Linha teórica, técnicas previstas, recursos terapêuticos)

**TEMAS PRIORITÁRIOS PARA TRABALHO**
(Ordenados por urgência/relevância clínica)

**FREQUÊNCIA E MODALIDADE**
(Proposta de atendimento)

**INDICADORES DE PROGRESSO**
(Como avaliar evolução)

**PRÓXIMOS PASSOS IMEDIATOS**
(O que será trabalhado nas próximas sessões)

{additional_context}

TRANSCRIÇÃO:
{transcript}""",
}
